src/datasources/balances-api/zerion-balances-api.service.ts

```diff
@@ -138,17 +134,18 @@ export class ZerionBalancesApi implements IBalancesApi {
           sort: 'value',
         },
       };
-      const { data } = await this.networkService.get<ZerionBalances>({
-        url,
-        networkRequest,
-      });
-      const zerionBalances = ZerionBalancesSchema.parse(data);
+      const zerionBalances = await this.networkService
+        .get<ZerionBalances>({
+          url,
+          networkRequest,
+        })
+        .then(({ data }) => ZerionBalancesSchema.parse(data));
       await this.cacheService.hSet(
         cacheDir,
         JSON.stringify(zerionBalances.data),
         this.defaultExpirationTimeInSeconds,
       );
-      return this._mapBalances(chainName, data.data);
+      return this._mapBalances(chainName, zerionBalances.data);
```

Comment:
Considering we intentially cache the response, it should be valid but we need it to be typed correct. Therefore it needs to either be casted or validated.

Validation has a positive and negative side effect: it'll catch incorrect responses at runtime at the cost of having to validate each time.

In general, I think it's safest to validate to get the type instead of casting it but we can cast if you think best?

Reply:
Thanks for the answer. Now it's more clear. As we usually validate, let's keep it as is for now.